File: app/src/main/java/ru/art2000/helpers/GeneralHelper.java
package ru.art2000.helpers;

import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.List;

public class GeneralHelper {

    public static NumberFormat resultNumberFormat = new DecimalFormat("#.#######");

    @SuppressWarnings("SameParameterValue")
    public static String joinToString(List list, String separator, String prefix, String postfix) {
        StringBuilder builder = new StringBuilder(prefix);
        int count = 0;
        for (Object obj : list) {
            if (++count > 1) {
                builder.append(separator);
            }
            builder.append(obj);
        }
        builder.append(postfix);
        return builder.toString();
    }

}

File: app/src/main/java/ru/art2000/helpers/GeneralHelper.kt
package ru.art2000.helpers

import java.lang.StringBuilder
import java.text.DecimalFormat
import java.text.NumberFormat

object GeneralHelper  {

    @JvmField
    val resultNumberFormat: NumberFormat = DecimalFormat("#.#######")

    @JvmStatic
    fun joinToString(list: List<*>, separator: String, prefix: String, postfix: String): String {
        val builder = StringBuilder(prefix)
        for ((count, obj) in list.withIndex()) {
            if (count + 1 > 1) {
                builder.append(separator)
            }
            builder.append(obj)
        }
        builder.append(postfix)
        return builder.toString()
    }
}
